Marlin/SanityCheck.h

```diff
@@ -292,7 +292,15 @@
   #endif
 
   /**
+   * Check if ENDSTOPS_ONLY_FOR_HOMING is disabled when M211 command is enabled
+   */
+  #if (defined(ENABLE_M211)) && (!defined(ENDSTOPS_ONLY_FOR_HOMING))
+    #error ENDSTOPS_ONLY_FOR_HOMING must be enabled when M211 command is active.
+  #endif
+
+  /**
    * Warnings for old configurations
+   * Check if ENDSTOPS_ONLY_FOR_HOMING is disabled when M211 command is enabled
```

Comment:
I want to use marlin for milling and printing with only one configuration. In printing mode i can home the Z axis automatically with an servo. In milling mode i need manual home the Z axis.

To do manual homing and milling down i need to allow moving Z under 0 position. I can allow this all the time or, with this command only when i need this.

I want to block moves under 0 when an extruder is mounted to protect my hardware when i send an milling file to marlin.

My first idea was to implement secondary homing (G30) but this command is defined with another functionality in marlin and needs to add an secondary pointer set for X,Y and Z.